packages/connect-form/src/components/connect-form.tsx

```diff
@@ -44,81 +53,165 @@ const formContentContainerStyles = css({
   padding: spacing[4],
 });
 
+const favoriteButtonStyles = css({
+  position: 'absolute',
+  top: spacing[4],
+  right: spacing[6],
+  hover: {
+    cursor: 'pointer',
+  },
+});
+
+const editFavoriteButtonStyles = css({
+  verticalAlign: 'text-top',
+  marginLeft: spacing[1],
+});
+
+const favoriteButtonLabelStyles = css({
+  position: 'absolute',
+  top: spacing[5],
+  paddingTop: spacing[1],
+  color: uiColors.black,
+  fontWeight: 'bold',
+  fontSize: 12,
+});
+
 function ConnectForm({
   initialConnectionInfo,
   onConnectClicked,
+  // The connect form will not always used in an environment where
+  // the connection info can be saved.
+  showSaveConnection = false,
+  saveConnection,
 }: {
   initialConnectionInfo: ConnectionInfo;
   onConnectClicked: (connectionInfo: ConnectionInfo) => void;
+  showSaveConnection?: boolean;
+  saveConnection: (connectionInfo: ConnectionInfo) => Promise<void>;
```

Comment:
if there is `showSaveConnection` should `saveConnection` be optional?

Reply:
Yup, made it just use one optional function for now.